src/Microsoft.Azure.WebJobs.Host/Utility.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Azure.WebJobs.Host.Config;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Azure.WebJobs
+{
+    internal class Utility
+    {
+        private const string DefaultConfigurationRootSectionName = "AzureWebJobs";
+        private const string ConfigurationRootSectionKey = "AzureWebJobsConfigurationSection";
+
+        public static IConfigurationSection GetWebJobsConfigurationSection(IConfiguration configuration)
+        {
+            string configPath = configuration[ConfigurationRootSectionKey] ?? DefaultConfigurationRootSectionName;
+
+            return configuration.GetSection(configPath);
+        }
+
+        public static IConfigurationSection GetExtensionConfigurationSection<T>(IConfiguration configuration)
+            where T : IExtensionConfigProvider
+        {
+            return GetExtensionConfigurationSection(configuration, GetExtensionName<T>());
+        }
+
+        public static IConfigurationSection GetExtensionConfigurationSection(IConfiguration configuration, string extensionName)
+        {
+            string configPath = "extensions";
+
+            if (extensionName != null)
```

Comment:
We'll allow a null extension name? Does that mean it tries to read from directly under "extensions"?